src/network/networkd-network.c

```diff
@@ -399,6 +399,18 @@ int network_apply(Manager *manager, Network *network, Link *link) {
         return 0;
 }
 
+bool network_has_static_ipv6_addresses(Network *network) {
+        Address *address;
+
+        assert(network);
+
+        while ((address = network->static_addresses))
+               if (address->family == AF_INET6)
+                       return true;
```

Comment:
This looks like an endless loop to me. You never take the next address, always just the first one...